### ISSUE TITLE
[feature] Add ``fstack-protector-strong`` gcc flag

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#10351]
It is proved that this flag: 
- doesnt change the ABI, it can be safely added
- provide more secure artifacts

The approach:
- add it to new generators: CMakeToolchain, AutootoolsToolchain, MesonToolchain
- It applies to ``gcc`` compiler in Linux systems (need to check if in all gcc versions)
- Probably we want to introduce a ``tools.build:secure_builds`` or similar ``[conf]`` that allows this feature to be opted-out. The name might be generic, because we might want to add more flags later, to other compilers

**Comment by brianmou** [comment]
` fstack-protector` should be added for gcc 4.1.0-4.8.5, and `fstack-protector-strong` for gcc >=4.9.0 and clang >= 5.0

**Comment by memsharded** [comment]
We are going for the following approach:

- Create secure profiles for compilers, that will define a [conf] section with CXX_FLAGS that define the secure flags
- This is installable with ``conan config install``
- ConanCenter profiles will use the secure profiles for generating packages

This needs to wait for some infra around [conf] feature and CXX_FLAGS general injection to different build systems.

**Comment by lasote** [comment]
We will start testing these flags in the fork of recipes of conan-center-index for Conan 2.0 and see how it goes. It might happen that recipes fail to build.

**Comment by franramirez688** [comment]
Let's recall that, since Conan 1.47, it should be enough to add these lines to your profile:
```
[conf]
tools.build:cxxflags+=["-fstack-protector-strong"]
```
Or even to the *global.conf*:
```
tools.build:cxxflags+=["-fstack-protector-strong"]
```
**Note**: the examples above are appending the value with `+=` but you can also redefine it using `=`.

**Comment by SSE4** [comment]
> Let's recall that, since Conan 1.47, it should be enough to add these lines to your profile:

it's C++ only, right? probably we also need:
```
[conf]
tools.build:cflags+=["-fstack-protector-strong"]
```
and maybe also:
```
[conf]
tools.build:objcflags+=["-fstack-protector-strong"]
tools.build:objcxxflags+=["-fstack-protector-strong"]
```
for Objective-C/Objective-C++ sources on macOS?

UPD: but anyway, I would vote for higher-level abstraction. this might not work for older GCC, and for sure it won't work for MSVC.

**Comment by memsharded** [comment]
> for Objective-C/Objective-C++ sources on macOS?

We have feedback to add them to Linux-gcc only at this moment, no need to add them in OSX

> UPD: but anyway, I would vote for higher-level abstraction. this might not work for older GCC, and for sure it won't work for MSVC.

The conf goes in the profile, together with the compiler and the compiler version. It is the responsibility of the profile definition to be correct, not adding it if the compiler or the compiler version doesn't support it.

> tools.build:cflags+=["-fstack-protector-strong"]

Yes, the flag should also apply to C

**Comment by SSE4** [comment]
so, why no `tools.build:secure_builds` as it was iniitially proposed?

**Comment by memsharded** [comment]
This feature is ready and available in 1.47, defined in profiles CXXFLAGS.
We can close it, next step is start to apply secure profiles in the creation of ConanCenter packages by default.
